compaction/table_state.hh

```diff
@@ -33,6 +34,8 @@ public:
     virtual bool compaction_enforce_min_threshold() const noexcept = 0;
     virtual const sstables::sstable_set& main_sstable_set() const = 0;
     virtual const sstables::sstable_set& maintenance_sstable_set() const = 0;
+    virtual const std::unordered_set<sstables::shared_sstable>& cleanup_sstable_set() const = 0;
+    virtual std::unordered_set<sstables::shared_sstable>& cleanup_sstable_set() = 0;
     virtual std::unordered_set<sstables::shared_sstable> fully_expired_sstables(const std::vector<sstables::shared_sstable>& sstables, gc_clock::time_point compaction_time) const = 0;
```

Comment:
cleanup_sstable_set is also not needed outside compaction_manager. 

I think _owned_ranges_ptr and cleanup sstable set will nicely go into compaction_manager's per table state. For example, list of sstables being compacted is not needed elsewhere, therefore it's managed by manager. I think the same applies to list of sstables pending cleanup.

By the way, cleanup_sstable_set can be renamed to sstables_requiring/needing_cleanup, which is more verbose but also more descriptive. To me, cleanup_sstable_set is not very easy to deduce its meaning.

Reply:
done in v2